commands/command_migrate_export.go

```diff
@@ -159,17 +159,17 @@ func migrateExportCommand(cmd *cobra.Command, args []string) {
 		ExitWithError(err)
 	}
 
-	// Prune the cache
-	for _, object := range exported {
-		err := os.Remove(object)
-		if os.IsNotExist(err) {
-			continue
-		}
+	fetchPruneCfg := lfs.NewFetchPruneConfig(cfg.Git)
 
-		if err != nil {
-			ExitWithError(err)
-		}
-	}
+	// Set our preservation time-window for objects existing on the remote to
+	// 0. Because the newly rewritten commits have not yet been pushed, some
+	// exported objects can still exist on the remote within the time window
+	// and thus will not be pruned from the cache.
+	fetchPruneCfg.PruneOffsetDays = 0
```

Comment:
I left a comment above but this was required to ensure certain objects were properly pruned. `prune` preserves any cached objects that exist on the remote from commits within a time window of `PruneOffsetDays + FetchRecentRefsDays`. Since `migrate export` rewrites commits but does not push them, exported objects can still exist on the remote, and can be retained in the cache if this time window is not set to 0.

A negative side effect of this is that _all_ objects that exist on the remote will be pruned, regardless of whether or not the associated object was exported. It seems like this should be okay though, as if necessary Git LFS can just re-download the pruned files.

Reply:
With respect to my comment below, what happens if we remove these assignments iff we perform the prune _after_ we do the checkout?

Reply:
These values only affect whether to preserve pointers referenced in the remote git history, so they're still required if the prune is moved after the checkout

Reply:
It looks like in general though we don't actually need to set `PruneOffsetDays`, just `FetchRecentRefsDays`, so I'll remove it